README.md

```diff
@@ -1,67 +1,153 @@
 
 - [OpenChat group](https://oc.app/2zyqk-iqaaa-aaaar-anmra-cai)
 - [Twitter](https://twitter.com/mr_research_ag)
 - [Dfinity forum](https://forum.dfinity.org/)
 
-### Motivation
+## Build & test
 
-The package was developed to streamline and secure the entire lifecycle of user funds on an ICRC1 ledger, including deposits, credits, and withdrawals.
+We need up-to-date version of `mops` installed.
 
-## Usage
+Then run:
+
+```
+git clone git@github.com:research-ag/token-handler.git
+mops install
+mops test
+```
 
-### Install with mops
+## Installation
 
-You need `mops` installed. In your project directory run:
+You need `mops` installed. In your project directory run:
 
 ```
 mops add token-handler
 ```
 
 In the Motoko source file import the package as:
 
-```motoko
+```
 import TokenHandler "mo:token-handler";
 ```
 
-### Example
+## Example
 
-To see token-handler package in action, check out the [example code](https://github.com/research-ag/token-handler/blob/main/example).
+To see token-handler package in action, check out the [example code](https://github.com/research-ag/token-handler/blob/main/example). The example is a simple implementation of the [ICRC-84](https://github.com/research-ag/icrc-84) standard.
 
-### Build & test
+## Documentation
 
-We need up-to-date versions of `node`, `moc` and `mops` installed.
-Suppose `<path-to-moc>` is the path of the `moc` binary of the appropriate version.
+### Requirements
 
-Then run:
+The only requirement on the underlying token ledger is the ICRC-1 standard. Deposit method via direct transfer is available without ICRC-2 extension. For using the deposit method via an allowance ICRC-2 extension is required.
 
-```
-git clone git@github.com:research-ag/token-handler.git
-mops install
-DFX_MOC_PATH=<path-to-moc> mops test
-```
+It is not required that the service can inspect individual deposit transactions by transaction id, memo or other means. Hence, it is not required that the underlying token ledger provides an indexer, transaction history or archive. In particular, the ICRC-3 extension is not required.
+
+### TokenHandler options
+
+| Property name | Type | Description |
+| --- | --- | --- |
+| ledgerApi | LedgerAPI | Object that represents the ledger API. |
+| ownPrincipal | Principal | Service principal. |
+| initialFee | Nat | Initial fee value. |
+| triggerOnNotifications | Bool | Flag that enables scheduling consolidation after a deposit notification. |
+| log | (Principal, LogEvent) -> () | Callback that is used to log events inside the token handler. |
+
+### Fees processing
+
+Fees are made up of two components: the surcharge and the ledger fee.
+
+Depending on the operation, fees are calculated as follows:
+
+| Operation | Fee value |
+| --- | --- |
+| Deposit via direct transfer | surcharge + ledger_fee |
+| Deposit via an allowance | surcharge |
+| Withdrawal from credit | surcharge + ledger_fee |
+| Withdrawal from pool | ledger_fee |
```

Comment:
The surcharge isn't in the code, is it? The TokenHandler allows the user to set arbitrary fees for each operation.